### PR TITLE
lib,tarball: document missing exported functions

### PR DESCRIPTION
**Pull request by iaguis**
Also, make license headers consistent.

### CODE REVIEW

File: tarball/tarfile.go [hunk 1]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Packege tarball provides functions to manipulate tar files.
```
Comment (krnowak) [inline tarball/tarfile.go]: Package

Comment (iaguis) [inline tarball/tarfile.go]: Ouch

**Comment by krnowak** [comment]
One typo spotted. I'm wondering if documentation of the `docker2aci` subpackages (`types`, `backend` and `util`) should mention that they are an implementation detail and should not be used. Dunno about `common` subpackage - it exports `Compression` enumeration which is used in the exported `docker2aci.CommonConfig` type.

File: lib/backend/file/file.go [hunk 2]
```diff
@@ -1,3 +1,19 @@
+// Copyright 2015 The appc Authors
```
Comment (alban) [inline lib/backend/file/file.go]: 2016?

Comment (iaguis) [inline lib/backend/file/file.go]: Welp, the code was written in 2015...

**Comment by iaguis** [comment]
> One typo spotted. I'm wondering if documentation of the docker2aci subpackages (types, backend and util) should mention that they are an implementation detail and should not be used. Dunno about common subpackage - it exports Compression enumeration which is used in the exported docker2aci.CommonConfig type.

I'll add a note for types, backend and util.

**Comment by iaguis** [comment]
Updated.

File: lib/common/common.go [hunk 3]
```diff
@@ -44,6 +59,17 @@ const (
 	GzipCompression
 )
 
+type ErrSeveralImages struct {
```
Comment (krnowak) [inline lib/common/common.go]: Docs missing.

**Comment by krnowak** [comment]
Consider splitting the `common` package into two - the exported `common` package and the `internal` package being an implementation detail.

File: tarball/tarfile.go [hunk 4]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Package tarball provides functions to manipulate tar files.
```
Comment (krnowak) [inline tarball/tarfile.go]: Mark it as implementation detail - if somebody complains, we can probably export it as a `pkg` subpackage?

Comment (iaguis) [inline tarball/tarfile.go]: Alright

**Comment by iaguis** [comment]
> Consider splitting the common package into two - the exported common package and the internal package being an implementation detail.

OK, I'll do that

**Comment by iaguis** [comment]
Updated.

File: lib/docker2aci.go [hunk 5]
```diff
@@ -37,11 +38,21 @@ import (
 	appctypes "github.com/appc/spec/schema/types"
 )
 
+// Docker2ACIBackend is the interface that abstracts converting Docker layers
```
Comment (krnowak) [inline lib/docker2aci.go]: This can be moved to the `internal` package now.

**Comment by krnowak** [comment]
Hopefully last nit.

**Comment by iaguis** [comment]
Hopefully...

**Comment by krnowak** [comment]
LFAD.

**Comment by iaguis** [comment]
\ø/

**Pull request merged**